### PR TITLE
review fix: Object[]::new references resolution

### LINKED ISSUE #1883: bug: Java8 "Foo[]::new" cannot be resolved
Assuming a Java8 `new ArrayList().stream().toArray(Foo[]::new)`

The `Foo[]::new` is a CtExRefExpression which cannot be resolved with `getExecutableDeclaration` even if Foo is in the classpath.

See failing test in https://github.com/INRIA/spoon/compare/master...monperrus:bug-execrefexpr

### PR DESCRIPTION
**Pull request by surli**
fix #1883

### CODE REVIEW

**Comment by surli** [comment]
The proposed solution is far from being perfect but it's a first step. I propose that before improving it further we define first a better test contract.

File: src/test/java/spoon/test/reference/ExecutableReferenceTest.java [hunk 1]
```diff
@@ -204,4 +207,23 @@ public void testHashcodeWorksWithReference() {
 
 		assertNotEquals(hashCode1, hashCode2);
 	}
+
+	@Test
+	public void testPbWithStream() {
+		// contract: array constructor references are well represented
```
Comment (monperrus) [inline src/test/java/spoon/test/reference/ExecutableReferenceTest.java]: add assert on type of element?
add assert on toString of element?

File: src/main/java/spoon/support/reflect/declaration/InvisibleArrayConstructor.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+package spoon.support.reflect.declaration;
+
+/**
+ * This class is used to represent the constructor of an array when calling with an expression like
+ * <pre>
+ *     String[]::new
+ * </pre>
+ */
+public class InvisibleArrayConstructor<T> extends CtConstructorImpl<T> {
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/declaration/InvisibleArrayConstructor.java]: By convention, should end by Impl: `InvisibleArrayConstructorImpl`, and should be in `spoon.support.reflect.declaration` (all implementation classes are in spoon.support)

**Comment by monperrus** [comment]
Thanks @surli.

LGTM, see comments.

File: src/main/java/spoon/support/reflect/declaration/InvisibleArrayConstructor.java (hunk 2 above)
Comment (surli) [reply src/main/java/spoon/support/reflect/declaration/InvisibleArrayConstructor.java]: It is in spoon.support. I will rename it.

**Comment by surli** [comment]
For me it's ready again.

**Comment by monperrus** [comment]
Thanks Simon!

**Pull request merged**